cmd/getResources.go

```diff
@@ -0,0 +1,114 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/equinor/radix-cli/generated-client/client/application"
+	"github.com/equinor/radix-cli/pkg/client"
+	"github.com/equinor/radix-cli/pkg/config"
+	"github.com/equinor/radix-cli/pkg/flagnames"
+	"github.com/equinor/radix-cli/pkg/utils/json"
+	"github.com/equinor/radix-common/utils/pointers"
+	"github.com/spf13/cobra"
+)
+
+// getResourcesCmd represents the getResourcesCmd command
+var getResourcesCmd = &cobra.Command{
+	Use:   "resources",
+	Short: "Gets resources used by the Radix application",
+	Long:  `Gets resources used by the Radix application or its environment or a component`,
+	Example: `
+# Get resources used by the application for the last 30 days
+rx get resources -a myapp
+
+# Get resources used by the application in the environment for the last 30 days
+rx get resources -a myapp -e dev
+
+# Get resources used by the application in the environment for a component for the last 30 days
+rx get resources -a myapp -e dev -n mycomponent
+
+# Get resources used by the application in the environment for a component for the last 5 minutes
+rx get resources -a myapp -e dev -n mycomponent --duration 5m
+
+# Get resources used by the application in the environment for a component for the last 12 hours starting with two days ago
+rx get resources -a myapp -e dev -n mycomponent --duration 12h --since 2d
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		appName, err := config.GetAppNameFromConfigOrFromParameter(cmd, flagnames.Application)
+		if err != nil {
+			return err
+		}
+		if appName == "" {
+			return errors.New("application name is required field")
+		}
+		envName, err := cmd.Flags().GetString(flagnames.Environment)
+		if err != nil {
+			return err
+		}
+		componentName, err := cmd.Flags().GetString(flagnames.Component)
+		if err != nil {
+			return err
+		}
+		since, err := cmd.Flags().GetString(flagnames.Since)
+		if err != nil {
+			return err
+		}
+		duration, err := cmd.Flags().GetString(flagnames.Duration)
+		if err != nil {
+			return err
+		}
+
+		getResourcesParams := application.NewGetResourcesParams()
+		getResourcesParams.SetAppName(appName)
+		getResourcesParams.SetEnvironment(&envName)
+		getResourcesParams.SetComponent(&componentName)
+		if duration != "" {
+			getResourcesParams.SetDuration(pointers.Ptr(duration))
+		}
+		if since != "" {
+			getResourcesParams.SetSince(pointers.Ptr(since))
+		}
+
+		cmd.SilenceUsage = true
+
+		apiClient, err := client.GetForCommand(cmd)
+		if err != nil {
+			return err
+		}
+		resp, err := apiClient.Application.GetResources(getResourcesParams, nil)
+		if err != nil {
+			return err
+		}
+		prettyJSON, err := json.Pretty(resp.Payload)
+		if err != nil {
+			return err
+		}
+		fmt.Println(*prettyJSON)
+		return nil
+	},
+}
+
+func init() {
```

Comment:
Include flag completion, see restartComponent.go.go for an example

Reply:
Added flag

Reply:
Missing flag completion for environment and component